lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3434,6 +3434,12 @@ namespace {
       result->setInterfaceType(type);
       result->setIsObjC(false);
       result->setIsDynamic(false);
+      if (decl->getType().isConstQualified()) {
+        // Note that in C++ there are ways to change the values of const
+        // members, so we don't use WriteImplKind::Immutable storage.
```

Comment:
What would the consequences of `Immutable` be?

Reply:
My guess is that it enables (mis)optimizations based on the (incorrect) assumption that those fields are never modified.